packages/extractor/src/extractor/mod.js

```diff
@@ -187,96 +182,36 @@ class Extractor {
 	 * @returns {ts.Program}
 	 */
 	#create_program() {
-		const root_names = [...create_default_root_names(), this.compiler.filepath];
 		const program = ts.createProgram({
-			rootNames: root_names,
-			options: this.#get_ts_options(),
+			rootNames: [this.#options.shims_path, this.#options.tsx_filepath],
+			options: this.#options.get_ts_options(),
 			host: this.#create_host(),
 			oldProgram: this.#cache.program,
 		});
 		return program;
 	}
 
-	/** @returns {ts.CompilerOptions} */
```

Comment:
I found these lines of code can be moved to the Option class, since they can be derived solely based on the filepath.